### ISSUE TITLE
creating a record calls different (or no??) serializer methods in 2.10+

### ISSUE THREAD

**Issue by fivetanley** [emberjs/data#4802]
Ember Twiddle: https://ember-twiddle.com/1811e61d99eeff347f2067e30b41c23a?openFiles=serializers.post.js%2C

Basically, we are using the `primaryKey` string in the serializer. **That** still works in 2.10 (but not in 2.11), however, no `normalize`/`normalizeResponse`/`normalizeCreateRecordResponse` hooks seem to be called in versions 2.10 and above. This includes canary. You can switch the different ember data versions in the twiddle.

Expected behavior:
- Using `primaryKey: 'some-non-id-primary-key'` should work. There shouldn't be an error complaining about IDs.
- You should be able to intercept the payload in `normalizeResponse` at the very least, if not `normalizeCreateRecordResponse`. One of those methods should be called (I have an `alert` that never gets hit).

Given a payload of 

```js
{
  post: {
    'some-non-id-primary-key': '1',
    'title': ' rails is omakase'
  }
}
```

And a serializer of:

```js
export default DS.RESTSerializer.extend({
  primaryKey: 'some-non-id-primary-key'
});
```

The record should:

```js
post.get('id') // '1'
post.get('title') // 'rails is omakase'
```

The `primaryKey` API does seem to work in all the GET variations (store.findRecord, store.find). This error only seems to be prevalent in the `createRecord` code path.

**Comment by fivetanley** [comment]
yeesh, i don't know how to mockjax.
